File: vmsh.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {
    // Change current directory to /mnt
    val cdProcess = ProcessBuilder("cd", "/mnt").start()
    cdProcess.waitFor()
    if (cdProcess.exitValue() != 0) {
        println("Error changing directory: ${cdProcess.getErrorStream().bufferedReader().readText()}")
        return
    }

    // Create hgfs directory if it doesn't exist
    val mkdirProcess = ProcessBuilder("sudo", "mkdir", "-p", "hgfs").start()
    mkdirProcess.waitFor()
    if (mkdirProcess.exitValue() != 0) {
        println("Error creating directory: ${mkdirProcess.getErrorStream().bufferedReader().readText()}")
        return
    }

    // Add entry to /etc/fstab file
    val echoCommand = "echo 'vmhgfs-fuse /mnt/hgfs fuse defaults,allow_other 0 0' | sudo tee -a /etc/fstab"
    val echoProcess = ProcessBuilder("/bin/sh", "-c", echoCommand).start()
    echoProcess.waitFor()
    if (echoProcess.exitValue() != 0) {
        println("Error adding entry to /etc/fstab: ${echoProcess.getErrorStream().bufferedReader().readText()}")
        return
    }

    // Reboot the system to apply changes
    val rebootProcess = ProcessBuilder("sudo", "reboot", "now").start()
    rebootProcess.waitFor()
    if (rebootProcess.exitValue() != 0) {
        println("Error rebooting system: ${rebootProcess.getErrorStream().bufferedReader().readText()}")
        return
    }

    println("Successfully created hgfs folder and added entry to /etc/fstab. System reboot initiated.")
}

File: vmsh.php
<?php

// Change current directory to /mnt
exec('cd /mnt', $output, $return_var);
if ($return_var !== 0) {
  echo 'Error changing directory: ' . implode("\n", $output);
  exit(1);
}

// Create hgfs directory if it doesn't exist
exec('sudo mkdir -p hgfs', $output, $return_var);
if ($return_var !== 0) {
  echo 'Error creating directory: ' . implode("\n", $output);
  exit(1);
}

// Add entry to /etc/fstab file
exec('sudo sh -c \'echo "vmhgfs-fuse /mnt/hgfs fuse defaults,allow_other 0 0" >> /etc/fstab\'', $output, $return_var);
if ($return_var !== 0) {
  echo 'Error adding entry to /etc/fstab: ' . implode("\n", $output);
  exit(1);
}

// Reboot the system to apply changes
exec('sudo reboot now', $output, $return_var);
if ($return_var !== 0) {
  echo 'Error rebooting system: ' . implode("\n", $output);
  exit(1);
}

echo 'Successfully created hgfs folder and added entry to /etc/fstab. System reboot initiated.';

?>
